src/views/SignUpScreen.jsx

```diff
@@ -72,43 +84,115 @@ export class SignUpScreen extends Component {
       <View style={SignUpStyle.container}>
         <ScrollView style={SignUpStyle.scroll}>
           <Text style={SignUpStyle.title}>Registrarse</Text>
-          <TextInputHandle
-            label="Nombre Completo"
-            placeholder="Nombre"
-            keylabel="inputValueName"
-            onChangeText={this.handleInputChange}
-          />
+          <Formik
+            initialValues={this.state.formikProps}
+            onSubmit={(values, formikHelpers) => {
+              formikHelpers.resetForm();
+            }}
+            validate={(values) => {
+              const emailValidRegex =
+                /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i;
+              const regexHood = /^[a-zA-Z]{4,20}$/;
+              const regex = /^[0-9]{5}$/;
+              const errors = {};
+              if (!values.name) {
+                errors.name = "Por favor, introduce un nombre";
+              } else if (values.name.length < 30) {
+                errors.name = "El nombre debe contener al menos 30 letras";
+              }
+              if (!values.hood) {
+                errors.hood = "Por favor, introduce tu locación";
+              } else if (!values.hood.match(regexHood) ? true : false) {
+                errors.hood =
+                  "La colonia debe tener al menos 4 letras y maximo 20";
+              }
+              if (!values.cp) {
+                errors.cp = "Por favor, introduce un código postal válido";
+              } else if (!values.cp.match(regex) ? true : false) {
+                errors.cp = "Utiliza los 5 números del código postal";
+              }
+
+              if (!values.email) {
+                errors.email = "Por favor, ingrese un correo electrónico";
+              } else if (errors.email == "auth/invalid-email") {
+                errors.email = "Texto para indicar que pongas tu mail";
+              } else if (
+                !(values.email.match(emailValidRegex) ? true : false)
+              ) {
+                errors.email = "Dirección de correo electrónico inválida";
+              }
+              if (!values.password) {
+                errors.password = "Por favor, ingrese una contraseña";
+              } else if (values.password.length < 8) {
+                errors.password =
+                  "La contraseña debe tener al menos 8 caracteres";
+              }
+
+              return errors;
+            }}
```

Comment:
It's not recommendable to fill your components with logic code, could you please move this to a function and add that function to this `validate` prop? You could put that function after line 79.